src/vmspawn/vmspawn.c

```diff
@@ -778,6 +846,22 @@ static int run_virtual_machine(void) {
                 }
         }
 
+        if (arg_qemu_initrd) {
+                initrd = strdup(arg_qemu_initrd);
+                if (!initrd)
+                        return log_oom();
+        } else if (arg_qemu_kernel && !endswith(arg_qemu_kernel, ".efi")) {
```

Comment:
i think that endswith() check really should move into find_initrd()

Reply:
Fixed in 37e75950ff.